fakestorage/bucket.go

```diff
@@ -14,9 +14,9 @@ import (
 // CreateBucket creates a bucket inside the server, so any API calls that
 // require the bucket name will recognize this bucket.
 //
-// If the bucket already exists, this method does nothing.
-func (s *Server) CreateBucket(name string) {
-	err := s.backend.CreateBucket(name)
+// If the bucket already exists, this method does nothing but panics if props differs
+func (s *Server) CreateBucket(name string, versioningEnabled bool) {
```

Comment:
This change is disruptive: we extend the previous contract with users of fakestorage requiring an extra arg (versioning enabled or not). If we dont want to introduce this breaking change, we can expose a CreateBucketWithVersioning, or something like CreateBucketWithOpts and the new arg (or accepting a struct that describes the Opts, that makes that more extensible), making the old (that can be marked as deprecated?) to call to the new one

Reply:
Let's do ``CreateBucketWithOpts`` and deprecate this one. That way we can extend it in the future too, if needed.

Reply:
Done, and updated the repo so we are using CreateBucketWithOpts everywhere. We could opt for a pattern a bit more sophisticated (https://github.com/tmrts/go-patterns/blob/master/idiom/functional-options.md), but at this stage, I think that may be enough. Is that fine for you?

Reply:
(reopen if you have further comments - this and any other comment im marking as resolved, as im using them to track remaining work from my side)